### DISCUSSION TITLE
CommandParameter doesn't work in ItemsControl.ItemTemplate

### DISCUSSION THREAD

**Discussion by Miticcio** [AvaloniaUI/Avalonia#9505] [Q&A] [answered]
**Describe the bug**
When creating an ItemsControl specifing an ItemTemplate, if you bind the CommandParameter property to a button inside the DataTemplate, the button is disabled because the binding fails and the Debug Console prints:
`[Binding]Error in binding to 'Avalonia.Controls.Button'.'Command': 'Could not find a matching property accessor for 'Duplicate' on 'TextCommandParameterInsideItemTemplate.ViewModels.MainWindowViewModel''`.
It works in Avalonia 0.10.18.

**To Reproduce**
1. Create a MVVM project with Avalonia 11.0-preview4.
2. Create a ObservableCollection<string> inside the MainWindowViewModel.
```
public ObservableCollection<string> Collection { get; set; } = new() { "Item1", "Item2", "Item3" };

public void Duplicate(string item)
{
    Collection.Add(item);
}

public void Add()
{
    Collection.Add("New Item");
}
```
3. Create a function with a string as parameter inside the MainWindowViewModel.
4. Create a function with any parameter inside the MainWindowViewModel.
5. Create a ItemsControl in the MainWindow.axaml Binding the collection and the functions.
```
<ItemsControl Items="{Binding Collection}">
  <ItemsControl.ItemTemplate>
    <DataTemplate>
      <StackPanel Orientation="Horizontal">
        <Button Content="{Binding .}"
	        Command="{Binding $parent[Window].DataContext.Duplicate}"
	        CommandParameter="Item" />
        <Button Content="{Binding .}"
	        Command="{Binding $parent[Window].DataContext.Duplicate}"
	        CommandParameter="{Binding .}" />
        <Button Content="{Binding .}"
	        Command="{Binding $parent[Window].DataContext.Add}" />
      </StackPanel>
    </DataTemplate>
  </ItemsControl.ItemTemplate>
</ItemsControl>
```
6. Only the last button works.

**Expected behavior**
All three buttons enabled and with a correct binding (it worked in 0.10.18).

**Desktop (please complete the following information):**

- OS: Windows 11 22H2
- Version 11.0.0-preview4

**Additional context**
Minimal reproduced projects:
[Avalonia-0.10.zip](https://github.com/AvaloniaUI/Avalonia/files/10058256/Avalonia-0.10.zip) - Avalonia 0.10.18 working example.
[Avalonia-11-preview4.zip](https://github.com/AvaloniaUI/Avalonia/files/10058257/Avalonia-11-preview4.zip) - Avalonia 11.0.0-preview4 not working.

**Comment by maxkatz6** [comment, marked as answer]
1. First button doesn't work, because your view model doesn't have `Duplicate(object)` method. That's a [breaking change](https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#binding-commands-to-methods) comparing to 0.10. You can't use strongly typed Duplicate(string) anymore.
2. Binding to command parameter never worked, duplicate of https://github.com/AvaloniaUI/Avalonia/issues/6840

**Reply by jp2masa** [reply]
To avoid future confusion: `CommandParameter` and `ConverterParameter` are different. `CommandParameter` is an `AvaloniaProperty`, so bindings should work.

**Comment by Miticcio** [comment]
Oh sorry, we didn't notice it was in the breaking changes. Thank you.

**Comment by timunie** [comment]
Converted into a discussion as other may encounter a similar issue.
